File: java/program/Petshop.java
// Kelas Petshop yang merepresentasikan entitas barang di petshop
class Petshop {
    private String id;        // ID unik dari barang
    private String nama;      // Nama barang
    private String stok;  // stok barang (misal: Makanan, Alat Pembersih, dll.)
    private String harga;     // Harga barang dalam bentuk string

    // Konstruktor default tanpa parameter
    Petshop() {
    }

    // Konstruktor dengan parameter untuk inisialisasi objek langsung
    public Petshop(String id, String nama, String stok, String harga) {
        this.id = id;
        this.nama = nama;
        this.stok = stok;
        this.harga = harga;
    }

    // Getter untuk mengambil nilai ID
    public String getId() {
        return id;
    } 

    // Setter untuk mengatur nilai ID
    public void setId(String id) {
        this.id = id;
    }

    // Getter untuk mengambil nama barang
    public String getNama() {
        return nama;
    }

    // Setter untuk mengatur nama barang
    public void setNama(String nama) {
        this.nama = nama;
    }

    // Getter untuk mengambil stok barang
    public String getStok() {
        return stok;
    }

    // Setter untuk mengatur stok barang
    public void setStok(String stok) {
        this.stok = stok;
    }

    // Getter untuk mengambil harga barang
    public String getHarga() {
        return harga;
    }

    // Setter untuk mengatur harga barang
    public void setHarga(String harga) {
        this.harga = harga;
    }
}

File: Petshop.php
<?php

// Mendefinisikan kelas Petshop untuk merepresentasikan item yang dijual di petshop
class Petshop{
    // Properti dengan akses private agar hanya bisa diakses melalui metode getter dan setter
    private $id;
    private $nama;
    private $stok;
    private $harga;
    private $urlfoto; // URL foto produk untuk menampilkan gambar produk

    // Konstruktor untuk menginisialisasi objek Petshop dengan nilai awal
    public function __construct($id, $nama, $stok, $harga, $urlfoto){
        $this->id = $id;
        $this->nama = $nama;
        $this->stok = $stok;
        $this->harga = $harga;
        $this->urlfoto = $urlfoto;
    }

    // Metode getter untuk mendapatkan nilai id
    public function getId() {
        return $this->id;
    }

    // Metode setter untuk mengubah nilai id
    public function setId($id) {
        $this->id = $id;
    }

    // Metode getter untuk mendapatkan nilai nama produk
    public function getNama() {
        return $this->nama;
    }

    // Metode setter untuk mengubah nilai nama produk
    public function setNama($nama) {
        $this->nama = $nama;
    }

    // Metode getter untuk mendapatkan stok produk
    public function getStok() {
        return $this->stok;
    }

    // Metode setter untuk mengubah stok produk
    public function setStok($stok) {
        $this->stok = $stok;
    }

    // Metode getter untuk mendapatkan harga produk
    public function getHarga() {
        return $this->harga;
    }

    // Metode setter untuk mengubah harga produk
    public function setHarga($harga) {
        $this->harga = $harga;
    }

    // Metode getter untuk mendapatkan URL foto produk
    public function getUrlFoto() {
        return $this->urlfoto;
    }

    // Metode setter untuk mengubah URL foto produk
    public function setUrlFoto($urlfoto) {
        $this->urlfoto = $urlfoto;
    }

    public function __destruct() {
    }  
}
?>